include/dsa/TypeSafety.h

```diff
@@ -78,6 +83,8 @@ struct TypeSafety : public ModulePass {
     // Methods for clients to use
     virtual bool isTypeSafe (const Value * V, const Function * F);
     virtual bool isTypeSafe (const GlobalValue * V);
+    virtual bool isFieldsOverlap(const Value * V, const Function *F);
```

Comment:
To be more consistent with the related `isTypeSafe` method, we should change the name and polarity of `isFieldsOverlap`. For example `isFieldDisjoint`, read “is field-disjoint”, sounds more consistent.